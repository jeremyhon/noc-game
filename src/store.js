import { createStore } from 'redux'
import rootReducer from './reducer'

const startStore = (initialState) => {
  return createStore(
    rootReducer,
    initialState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
}

const store = startStore()

export default store
