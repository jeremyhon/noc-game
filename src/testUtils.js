import { mount, shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';

// enzyme MUI Test Helpers
// - https://github.com/callemall/material-ui/issues/4664

const muiTheme = getMuiTheme();

/**
* MuiMountWithContext
*
* For `mount()` full DOM rendering in enzyme.
* Provides needed context for mui to be rendered properly
* during testing.
*
* @param {obj}    node - ReactElement with mui as root or child
* @return {obj}   ReactWrapper (http://airbnb.io/enzyme/docs/api/ReactWrapper/mount.html)
*/
export const muiMount = node => mount(node, {
  context: { muiTheme },
  childContextTypes: { muiTheme: PropTypes.object },
});

/**
* MuiShallowWithContext
*
* For `shallow()` shallow rendering in enzyme (component only as a unit).
* Provides needed context for mui to be rendered properly
* during testing.
*
* @param {obj}     node - ReactElement with mui
* @return {obj}    ShallowWrapper (http://airbnb.io/enzyme/docs/api/ShallowWrapper/shallow.html)
*/
export const muiShallow = node => shallow(node, {
  context: { muiTheme },
  childContextTypes: { muiTheme: PropTypes.object },
});

// touch tap event Helpers

export const simulateTouchTap = (element) => {
  const node  = ReactDOM.findDOMNode(element.node)
  TestUtils.Simulate.touchTap(node)
}
