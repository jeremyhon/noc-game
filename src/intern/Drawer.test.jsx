import React from 'react'
import { Provider } from 'react-redux'

import generateCompany from 'lib/company'
import { muiMount, muiShallow } from '../testUtils'
import { startStore } from '../store'
import ConnectedDrawer, { Drawer } from './Drawer'

let component

describe('Unconnected Drawer', () => {
  let props

  beforeEach(() => {
    props = {
      companies: [generateCompany()],
      deselectIntern: jest.fn(),
      selectedInternId: ""
    }
  })

  it("matches the snapshot", () => {
    component = muiShallow(<Drawer {...props} />)
    expect(component).toMatchSnapshot()
  })

  it("closes the drawer", () => {
    component = muiMount(<Drawer {...props} />)
    component.instance().changeDrawerState(false, 'swipe')
    expect(props.deselectIntern).toHaveBeenCalled()
  })
})

describe('ConnectedDrawer', () => {
  let store;
  beforeEach(() => {
    store = startStore()
  })

  it("matches the snapshot", () => {
    component = muiShallow(<ConnectedDrawer store={store}/>)
    expect(component).toMatchSnapshot()
  })

  describe("with provider", () => {
    beforeEach(() => {
      jest.spyOn(store, 'dispatch')
      component = muiMount(
        <Provider store={store}>
          <ConnectedDrawer />
        </Provider>
      )
    })

    it("opens the drawer", () => {
      expect(component.find("Drawer").at(1).props().open).toEqual(false)

      store.dispatch({type: 'SELECT_INTERN', payload: "abc"})

      expect(component.find("Drawer")).toBePresent()
      expect(component.find("Drawer").at(1).props().open).toEqual(true)
    })
  })
})
