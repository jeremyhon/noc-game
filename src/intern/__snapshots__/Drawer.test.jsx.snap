// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedDrawer matches the snapshot 1`] = `
<Drawer
  companies={Array []}
  deselectIntern={[Function]}
  selectedInternId=""
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;

exports[`Unconnected Drawer matches the snapshot 1`] = `
<Drawer
  className="drawer"
  disableSwipeToOpen={false}
  docked={false}
  onRequestChange={[Function]}
  open={false}
  openSecondary={true}
  swipeAreaWidth={30}
  width={270}
  zDepth={2}
/>
`;
